Classes/BITChannel.m

```diff
@@ -229,11 +234,23 @@ - (void)appendDictionaryToJsonStream:(NSDictionary *)dictionary {
     
     // Since we can't persist every event right away, we write it to a simple C string.
     // This can then be written to disk by a signal handler in case of a crash.
-    bit_appendStringToSafeJsonStream(string, &(BITSafeJsonEventsString));
+    bit_appendStringToBuffer(string);
     self.dataItemCount += 1;
   }
 }
 
+void bit_appendStringToBuffer(NSString *string) {
+  pthread_mutex_lock(&BITEventsStringMutex);
+  bit_appendStringToSafeJsonStream(string, &BITSafeJsonEventsString);
+  pthread_mutex_unlock(&BITEventsStringMutex);
+}
+
+void bit_resetBuffer(void) {
+  pthread_mutex_lock(&BITEventsStringMutex);
+  bit_resetSafeJsonStream(&BITSafeJsonEventsString);
+  pthread_mutex_unlock(&BITEventsStringMutex);
+}
+
 void bit_appendStringToSafeJsonStream(NSString *string, char **jsonString) {
```

Comment:
🔧 Maybe it's worth renaming the functions here since it turns out this is not actually safe? 🙂